### DISCUSSION TITLE
push new plugin to github

### DISCUSSION THREAD

**Discussion by randytry** [fwupd/fwupd#7047] [Q&A] [answered]
1.  Fork a project from fwupd github 
2.  Create my_new_changed branch
3.  Clone the project on my local branch and switch to the branch my_new_changed 
4.  Write a new plugin in local and commit on my local repository.
5.  Try use git push to my github 
6.  But I run test-fwupd and all tests are ok

In step 5 I found some issue below:

Username for 'https://github.com': chenhn123
Password for 'https://chenhn123@github.com': 
check for added large files..............................................Passed
check that executables have shebangs.................(no files to check)Skipped
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
black................................................(no files to check)Skipped
run tests before pushing.................................................Failed
- hook id: run-tests
- exit code: 

Ok:                 43  
Expected Fail:      0   
Fail:               2   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /home/randy/work/fwupd_my_new_branch/fwupd/venv/build/meson-logs/testlog.txt
FAILED: meson-internal__test 
/usr/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.

error: failed to push some refs to 'https://github.com/chenhn123/fwupd.git'

`

**Comment by superm1** [comment]
It looks like something is wrong with tests on your machine (which was enabled by your decision to opt in during contrib/setup).

Look at that log file for details on what test failed.

You can also run `test-fwupd` manually in the fwupd development environment to see the tests again.

**Reply by randytry** [reply]
[testlog.txt](https://github.com/fwupd/fwupd/files/14905149/testlog.txt)
[testlog_fail.txt](https://github.com/fwupd/fwupd/files/14905150/testlog_fail.txt)

I tried test-fwupd -> log is testlog.txt
and
git push  -> log is testlog_fail.txt

**Reply by randytry** [reply]
I don't know why same code base has different result

**Comment by superm1** [comment]
Here's the first problem:
```
11:07:26.621 FuHistory            building initial database
11:07:26.621 FuHistory            failed to migrate /home/randy/work/fwupd_my_new_branch/fwupd/venv/build/state/pending.db database: Failed to prepare SQL for creating tables: database is locked
11:07:26.621 FuHistory            trying to open database '/home/randy/work/fwupd_my_new_branch/fwupd/venv/build/state/pending.db'
Failed to load daemon: failed to load engine: Failed to prepare SQL for creating tables: disk I/O error
11:07:26.627 GLib                 g_atomic_ref_count_dec: assertion 'old_value > 0' failed
```

@hughsie think this is because of parallel tests perhaps?

**Comment by superm1** [comment]
Here's the second failure:
```
ok 22 /fwupd/security-attrs{compare}
# GLib-DEBUG: setenv()/putenv() are not thread-safe and should not be used after threads are created
# FuCommon-DEBUG: creating path /tmp/fwupd-self-test/etc/fwupd
not ok /fwupd/config - FuSelfTest:ERROR:../../libfwupdplugin/fu-self-test.c:847:fu_config_func: assertion failed (ret == 0): (-1 == 0)
Bail out!
----------------------------------- stderr -----------------------------------
**
FuSelfTest:ERROR:../../libfwupdplugin/fu-self-test.c:847:fu_config_func: assertion failed (ret == 0): (-1 == 0)
==============================================================================
```

Which I also wonder if it's because of parallel tests.

By chance were you running `test-fwupd` in one tab at the same time as trying to commit in another?

**Reply by randytry** [reply]
No, but I reboot my pc again and reproduce this issue
first git push 
 and I get the issue:
Ok:                 45  
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /home/randy/work/fwupd_my_new_branch/fwupd/venv/build/meson-logs/testlog.txt

error: failed to push some refs to 'https://github.com/chenhn123/fwupd.git'
If I try push again it will have test fail msg

**Comment by superm1** [comment, marked as answer]
If you're not actually running two instances of the test suite, I suggest for now you run this to turn off pre-push tests until we understand this.
```
# pre-commit uninstall -t pre-push
```

**Reply by randytry** [reply]
I push it on my company's github
https://github.com/chenhn123/fwupd/tree/my_new_change

**Reply by superm1** [reply]
OK;  open a pull request to upstream when you're ready.
